VENTouchLock/VENTouchLock.m

```diff
@@ -233,16 +226,39 @@ - (void)lock
                 }
                 [snapshotDisplayController loadView];
                 [snapshotDisplayController viewDidLoad];
-                snapshotDisplayController.view.frame = mainWindow.bounds;
-                self.snapshotView = snapshotDisplayController.view;
-                [mainWindow addSubview:self.snapshotView];
+                snapshotView = snapshotDisplayController.view;
             }
-            dispatch_async(dispatch_get_main_queue(), ^{
-                self.backgroundLockVisible = YES;
-                [rootViewController presentViewController:displayController animated:NO completion:nil];
-            });
         }
+    } else {
+        VENTouchLockEnterPasscodeViewController *enterPasscodeViewController = [[VENTouchLockEnterPasscodeViewController alloc] initWithTouchLock:self];
+        enterPasscodeViewController.didFinishWithResult = ^(BOOL success) {
+            __strong typeof(self) strongSelf = weakSelf;
+            strongSelf.locked = NO;
```

Comment:
since there are no guarantees this will be called on the main thread, let's wrap this in a `dispatch_async(dispatch_get_main_queue()..`